(SOL-531)(SOL-532) Course Cards update on the homepage;

marjev opened the pull request:
A link to course discovery page has been added to the OpenEdX homepage.

@marcotuts I would like you to take a look at the css I've redone

FYI
@martynjames @dsego

edx-webhook commented:
Thanks for the pull request, @marjev! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request. 

To automatically create an OSPR issue for this pull request, just visit this link: http://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-platform&number=7548

martynjames commented on lms/djangoapps/branding/tests/test_page.py:
```diff
@@ -203,8 +203,8 @@ def test_course_cards_sorted_by_default_sorting(self):
         self.assertEqual(template, 'index.html')
 
         # Now the courses will be stored in their announcement dates.
```
I *think* comment should say "sorted" not "stored"

martynjames commented on lms/djangoapps/branding/tests/test_page.py:
```diff
@@ -214,22 +214,22 @@ def test_course_cards_sorted_by_default_sorting(self):
         self.assertEqual(template, 'courseware/courses.html')
 
         # Now the courses will be stored in their announcement dates.
```
I *think* comment should say "sorted" not "stored". Also, I think that we should update the comment to reflect why the order has changed

martynjames commented on lms/djangoapps/branding/tests/test_page.py:
```diff
@@ -214,22 +214,22 @@ def test_course_cards_sorted_by_default_sorting(self):
         ((template, context), _) = RENDER_MOCK.call_args  # pylint: disable=unpacking-non-sequence
         self.assertEqual(template, 'index.html')
 
         # now the courses will be sorted by their creation dates, earliest first.
```
We've switched to order, but the comment says differently

martynjames commented on lms/static/sass/base/_grid-settings.scss:
```diff
@@ -7,7 +7,7 @@ $border-box-sizing: false;
 /* Breakpoints */
-$mobile: new-breakpoint(max-width 320px 4);
-$tablet: new-breakpoint(min-width 321px max-width 768px, 8);
+$mobile: new-breakpoint(max-width 540px 4);
```
Why are we changing these? Seems like the right thing - perhaps we really want to ADD a breakpoint for $large-mobile to reflect this - 320px is still where the iPhone comes into play (@talbs - any opinion...?)

talbs replied:
I'd also like to know the reason for this change. From the diff, I'm assuming that there's a need for a  new 540px - 768px range to support some layout elements. If this is the case, I'd recommend not adjusting the base ``$mobile`` breakpoint, but adding another breakpoint  in between ``$mobile`` and ``$tablet``. Perhaps something like this:

```scss
$mobile: new-breakpoint(max-width 320px 4);
$small-tablet: new-breakpoint(min-width 321px max-width 540px, 4);
$tablet: new-breakpoint(min-width 541px max-width 768px, 8);
```

Also, a few thoughts about this new addition:

1. You are changing the column count from 8 (what the previous ``$tablet`` breakpoint defined) to 4 for the range of 321px - 540px. This amy affect other UI elements that are assuming the 8 column layout. Please proof these (I believe @AlasdairSwan added these breakpoints and only the header/footer UI elements use them now).

2. This naming convention is getting a bit hairy and not handling cases like this well. I'd recommend that we move to changing these names/sizes to our emerging standards (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#responsive-breakpoints) so that they are device agnostic.

Hope that helps. Please ping me if you have any questions. I'll be reviewing this PR in more detail shortly.

marjev replied:
I've made these changes so that breakpoints in here correspond to the breakpoints used on the edx.org site.
The breakpoints used on the edx.org site are:
``` scss
// breakpoints for course cards
$bp-xl-min: 1180px;
$bp-l-min: 980px;
$bp-m-min: 768px;
$bp-s-min: 320px;
$bp-s-max: 320px;
$bp-m-max: 768px;
$bp-one-col-min: 540px;
```

martynjames replied:
@marjev - the example that you give above still includes a breakpoint at 320px. I really think we must add a new breakpoint if necessary instead of changing an existing one.

talbs replied:
I agree with @martynjames on this one - @marjev, can you add an additional breakpoint to this list for your case of 321px - 540px and re-instate the original mobile one?

Also, extra bonus points for renaming them to the sustainable system I mentioned above.

martynjames commented on lms/templates/index.html:
```diff
@@ -53,13 +53,18 @@
-            <ul class="courses-listing">
-            %for course in courses:
-              <li class="courses-listing-item">
+            <div class="courses-listing">
```
why the move to divs instead of `ul` and 'li`? It really is an list of things...

martynjames commented:
Please addresses quality failures

edx-webhook commented:
Thanks for the pull request, @marjev! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request. 

To automatically create an OSPR issue for this pull request, just visit this link: http://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-platform&number=7548

martynjames commented:
:+1: Looks okay to me, other than the style concerns. I understand that @talbs is going to look at styles more broadly to review these changes

martynjames commented on lms/djangoapps/courseware/courses.py:
```diff
@@ -383,7 +383,8 @@ def sort_by_start_date(courses):
     """
-    courses = sorted(courses, key=lambda course: (course.start is None, course.start), reverse=False)
+    courses = sorted(courses, key=lambda course: (
+        course.has_ended(), course.start is None, course.start), reverse=False)
```
Nit: This line-break formatting is a little strange, I'd prefer something along the lines of:
```
    courses = sorted(
        courses,
        key=lambda course: (course.has_ended(), course.start is None, course.start),
        reverse=False
    )
```

martynjames commented:
@marjev - The images in the cards are stretched / compressed to fit in the available space - this get particularly bad as one goes through the transitions. Try setting up few courses with different sizes and dimensions of images, and you should very quickly see the problem.

![screen shot 2015-04-02 at 3 32 52 pm](https://cloud.githubusercontent.com/assets/675165/6971901/d40604e6-d94d-11e4-8210-187a5657c1df.png)

![screen shot 2015-04-02 at 3 33 19 pm](https://cloud.githubusercontent.com/assets/675165/6971907/d79e30ec-d94d-11e4-9374-303fad6074a9.png)


Consider how the course images are handled in the dashboard page:
* Each image is constrained to the width of it's container - this is good, because one sees the full width the whole time. Only the top part of the image is visible. _*This is probably all that you need to change for this card view*_
* In the dashboard view, the relative heights of the images can increase as space allows _*I am not suggesting that this is right for the card view, given that each card should not increase in height - just highlighting it as a good example*_

marjev commented:
test this please

talbs commented:
@marjev, is there a sandbox I can take a look at these changes on? I'd love to check the rendered views with course data. If there's not an existing one, do you mind setting one up?

talbs commented on lms/static/sass/base/_animations.scss:
```diff
@@ -258,6 +258,18 @@
   }
 }
 
+@mixin fade-transition {
```
While this file might be old and already misused, we tend to keep this type of partial for just CSS and keyframe-based animations. It looks like you're wanting to just define how an element's visual state transitions. For that I'd recommend:

1. Using Bourbon's Transition mixin utility (http://bourbon.io/docs/#transitions) to handle prefixing (e.g. ``@include transition(all 0.2s ease-out);``)
2. Moving the Bourbon-based mixing reference to where you're calling it (e.g. the course image rule).

You can remove these definitions once that's done. In case you need to use this partial in the future for CSS/Keyframe animations, I'd recommend using Bourbon's ``animation()`` mixin - http://bourbon.io/docs/#animation.

talbs commented on lms/static/sass/base/_animations.scss:
```diff
@@ -258,6 +258,18 @@
 
+@mixin fade-transition {
+  -webkit-transition: all 0.2s ease-out;
+    -moz-transition: all 0.2s ease-out;
```
Please using the predefined timing variables (https://github.com/edx/edx-platform/blob/mjevtic/SOL-531/lms%2Fstatic%2Fsass%2Fbase%2F_variables.scss#L46) when setting a transition time.

talbs replied:
Also, be sure you want to transition all CSS property changes here. This will find any diffs in properties on elements you're calling this from.

talbs commented on lms/static/sass/base/_base.scss:
```diff
@@ -130,7 +130,6 @@ a:focus {
   margin: 0 auto 0;
   padding: ($baseline*2) 0;
   max-width: grid-width(12);
-  min-width: 760px;
```
@marjev, this may be related to the breakpoints and responsive scale you've altered elsewhere, but can you confirm:

1. Why you removed this value?
2. That you've visually proofed that this min-width isn't needed elsewhere in the platform (Courseware, Discussions, Login, etc.)?

marjev replied:
@talbs 

1. To make the section in which the course cards are contained responsive. Problem occurred for screen sizes lesser than 760px.
2. I will resolve this by adding more specific class for this section, something like `courses-container`. That way, I will be sure that I didn't break anything else in the platform while adjusting the section that I want to make responsive.

talbs replied:
Thanks, @marjev. Yeah, I would override this property on your view-specific Sass file (rather than in the _base.scss file). I'd recommend creating new Sass partials in /lms/static/sass/views/ for the homepage as well as the course results views. Anything specific to those can be contained in there and scoped to selectors those views have (i.e. ``.container-course``).

marjev replied:
@talbs,
would it be better if this was included in `/lms/static/sass/shared/_course_object.scss` since I'll be defining style rules that will be included in homepage, find courses page and course results page (last two will probably be the same page)?
`/lms/static/sass/shared/_course_object.scss` already defines style rules for every element that is nested within the section. I think that it would be easier to maintain it if we only had one file that defines style for the section which has course cards nested within. Any thoughts or suggestions?

talbs replied:
@marjev, hmmm. Looking at https://github.com/edx/edx-platform/blob/master/lms%2Fstatic%2Fsass%2Fshared%2F_course_object.scss, the styling here looks to be related to the very old homepage/course discovery views. In the short term, I think its fine to move this specific container-based override into that file, but you'll still need to scope it to a specific view (either by a specific class only used by this view - e.g. ``container-course`` or by assigning a view/layout-based class to the ``<body>`` element).

talbs commented on lms/static/sass/base/_base.scss:
```diff
@@ -130,7 +130,6 @@ a:focus {
   padding: ($baseline*2) 0;
   max-width: grid-width(12);
-  min-width: 760px;
   width: flex-grid(12);
```
ASIDE: Should this be removed if we're relying on Neat now to define our grid? @AlasdairSwan, was there a reason why you didn't remove this when adding in the Neat breakpoints and configuration (like the ``max-width: grid-width(12)`` rule above this one)?

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -7,33 +7,44 @@
 
     .courses-listing {
+      @include outer-container;
       @include clearfix();
```
I don't believe the ``clearfix()`` mixin is needed anymore with the ``outer-container`` being applied (I think it clears the document flow as part of its role). Please remove.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -7,33 +7,44 @@
-        width: flex-grid(4);
-        @include margin-right(flex-gutter());
-        @include float(left);
+        height: 360px;
```
Why the fixed height here? Is it to reinforce all course cards have the same height? If so, I'd recommend doing:

1. setting  a ``max-height`` instead
2. using the $baseline variable (our base unit of vertical/horizontal spacing) to help calculate this value (e.g. ``max-height: ($baseline*18)``)
3. creating Sass variables that you define at the top of this file, including one that is for ``$course-card-height`` and referencing it here.

marjev replied:
Your assumption was correct. I used it to make sure that all course cards have the same height. I will address this by applying your suggestions.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -7,33 +7,44 @@
+
+        @include media($mobile) {
+          @include span-columns(4);
+          margin: 15px 0px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable (e.g. ``($baseline*0.75)``).

talbs replied:
NIT: please do not define a unit of measurement on anything with a value of "0". Please change to ``margin: ($baseline*0.75) 0;``.

talbs replied:
Reviewing again now - are these left/right margins needed assuming we're using Neat's ``$gutter`` (http://thoughtbot.github.io/neat-docs/latest/#gutter)? This should give any elements assigned a grid width proportional left/right margins. I'd recommend relying on those rather than setting our own if possible.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -7,33 +7,44 @@
+
+        @include media($tablet) {
+          @include span-columns(4);
+          margin: 15px 3px 15px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable (e.g. ``($baseline*0.75)``). The 3px bottom margin can be left as is.

talbs replied:
Redundant margin-top and margin-bottom rules here (your base rules should be inherited and cover this case). Please remove these.

talbs replied:
RTL Support (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#right-to-left-rtl-support) is needed here for your margin-right property.

talbs replied:
See [my recent note about using gutter for margins](https://github.com/edx/edx-platform/pull/7548/files#r27965623).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -7,33 +7,44 @@
-        @include float(left);
+        height: 360px;
+
+        @include media($mobile) {
```
I'd prefer that we think of our rules as mobile-first (defining mobile as the base rules and then extending/overriding things like layout for larger breakpoints). In this case, that means moving the columns and margins-based rules to the base definition, like so: 

````scss
.courses-listing-item {
    @include margin-right(flex-gutter());
    @include float(left);
    height: 360px;
    @include span-columns(4);
    margin: 15px 0px;

    @include media($tablet) {
        ...
    }

    @include media($desktop) {
        ...
    }

    @include media($xl-desktop) {
        ...
    }
}
````

marjev replied:
@talbs I believe that 
````scss 
@include span-columns(4); 
````
should be within each breakpoint scope because its relative to the grid settings defined within each breakpoint variable. It is a coincidence that the parameter value is 4 for each variable, but absolute values that are a result of sass processing will be different since $mobile grid has only 4 columns, while $tablet grid has 8 columns, meaning that for mobile grid we want the course cards to occupy the entire grid, while for tablets we want them to occupy only one half of the grid (so that two course cards are presented in a row).

talbs replied:
@marjev, thanks for the reality check. My apologies, I misspoke there. Here are more thoughts:

* I'm with you on having Neat/Sass do the math for the width an element should take up given the ``$span`` value and the grid width for the current browser size/viewport.
* What was trying to get across in my previous (somewhat off-base) comment was that anything that is shared across viewports or needed by the default mobile/smallest viewport should be included outside of any media-query-based rules.
* If an element's width is the full width of the grid, I'd prefer we use the ``fill-parent()`` utility (http://thoughtbot.github.io/neat-docs/latest/#fill-parent) rather than defining the max number of columns of the grid at that breakpoint (since those column configurations can change over time).

With that said, my previous code example would look something like this:

```scss
.courses-listing-item {
    @include fill-parent();
    height: 360px; // please fix with above comments
    margin: 15px 0px;

    @include media($tablet) {
        @include span-columns(4); // 4 of 8
        ...
    }

    @include media($desktop) {
        @include span-columns(4); // 4 of 12
        ...
    }

    @include media($xl-desktop) {
        @include span-columns(3); // 3 of 12
        ...
    }
}
```

Hope that helps! Let me know if there's anything else confusing or just plain wrong about this.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -7,33 +7,44 @@
-          @include margin-right(0);
+        @include media($desktop) {
+          @include span-columns(4);
+          margin: 15px 5px 15px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable (e.g. ``($baseline*0.75)``).

talbs replied:
Redundant margin-top and margin-bottom rules here (your base rules should be inherited and cover this case). Please remove these.

talbs replied:
RTL Support (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#right-to-left-rtl-support) is needed here for your ``margin-right`` property.

talbs replied:
See [my recent note about using gutter for margins](https://github.com/edx/edx-platform/pull/7548/files#r27965623).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -7,33 +7,44 @@
+
+        @include media($xl-desktop) {
+          @include span-columns(3);
+          margin: 15px 7px 15px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable (e.g. ``($baseline*0.75)``).

talbs replied:
Redundant margin-top and margin-bottom rules here (your base rules should be inherited and cover this case). Please remove these.

talbs replied:
See [my recent note about using gutter for margins](https://github.com/edx/edx-platform/pull/7548/files#r27965623).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -7,33 +7,44 @@
+          @include span-columns(3);
+          margin: 15px 7px 15px;
+        }
+
```
NIT: remove extra space.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -7,33 +7,44 @@
       background: $body-bg;
-      border: 1px solid $border-color-1;
-      border-radius: 2px;
+      border-bottom: 3px solid #3185bd;
```
Please use or define a color Sass variable (in the ``_variables.scss`` file) for this type of color.

talbs replied:
NIT: property order is out of order (see https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#property-and-declaration-order)

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -7,33 +7,44 @@
-      border-radius: 2px;
+      border-bottom: 3px solid #3185bd;
       @include box-sizing(border-box);
       box-shadow: 0 1px 10px 0 rgba(0,0,0, 0.15), inset 0 0 0 1px rgba(255,255,255, 0.9);
```
Not your doing, but please convert these color values to the known transparent Sass variables - https://github.com/edx/edx-platform/blob/mjevtic/SOL-531/lms%2Fstatic%2Fsass%2Fbase%2F_variables.scss#L61 and https://github.com/edx/edx-platform/blob/mjevtic/SOL-531/lms%2Fstatic%2Fsass%2Fbase%2F_variables.scss#L70).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+      .course-image {
         .cover-image {
-          height: 200px;
+          height: 160px;
```
Why the fixed height here? Is it to reinforce all course images have the same height? If so, I'd recommend doing:

1. setting  a ``max-height`` instead
2. using the $baseline variable (our base unit of vertical/horizontal spacing) to help calculate this value (e.g. ``max-height: ($baseline*18)``)
3. creating Sass variables that you define at the top of this file, including one that is for ``$course-image-height`` and referencing it here.

marjev replied:
@talbs,
I've fixed the height so that the image container has the same height, no matter the height of the image. Setting the `max-height` as the alternative would take care of all the cases in which the image height is greater than 160px, while images with height lesser than 160px would cause the container to shrink to the height of the image it's containing.

Since we want to preserve the height of the image and make the course cards that are all of the same height, I've decided to take this approach, but I'm open to any suggestions.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-          width: 100%;
+
+          &:before {
+            @include fade-transition();
```
See previous comment about transitions and update this line.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            opacity: 0;
+            width: 100%;
+            height: 160px;
+            z-index: 10;
```
Please use one of our defined UI-depth Sass extends here (for easier UI management down the road) - https://github.com/edx/edx-platform/blob/mjevtic/SOL-531/lms%2Fstatic%2Fsass%2Fbase%2F_mixins.scss#L93

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            position: absolute;
+            top: 0;
+            left: 0;
+            background: black;
```
Please use the Sass ``$black`` variable - https://github.com/edx/edx-platform/blob/mjevtic/SOL-531/lms%2Fstatic%2Fsass%2Fbase%2F_variables.scss#L60

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-        width: 100%;
-        overflow: hidden;
-
+      .course-image {
```
NIT: property order is out of order (see https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#property-and-declaration-order)

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
           overflow: hidden;
-          width: 100%;
+
+          &:before {
```
Please add a Sass comment above this rule about what the pseudo element rule is doing (for future contributors to follow).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            z-index: 10;
+          }
+
+          &:after {
```
Please add a Sass comment above this rule about what the pseudo element rule is doing (for future contributors to follow).

talbs replied:
NIT: property order is out of order (see https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#property-and-declaration-order)

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+          }
+
+          &:after {
+            @include fade-transition();
```
See previous comment about transitions and update this line.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+          &:after {
+            @include fade-transition();
+            @include box-sizing(border-box);
+            @include calc( top, '80px - 25px');
```
Please add an inline Sass comment about what this math is doing and why it is needed.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            @include fade-transition();
+            @include box-sizing(border-box);
+            @include calc( top, '80px - 25px');
+            @include calc( left, '50% - 100px');
```
Please add an inline Sass comment about what this math is doing and why it is needed.

talbs replied:
Also, please use any Sass variables (either ``$baseline`` or specific course-card/image dimensions) you can.

talbs replied:
RTL Support (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#right-to-left-rtl-support) is needed here.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            @include box-sizing(border-box);
+            @include calc( top, '80px - 25px');
+            @include calc( left, '50% - 100px');
+            content: 'LEARN MORE';
```
I don't think this type of content should be rendered by the CSS layer or things. If its a call to action, we should make sure its:

* accessible (adding @cptvitamin to chime in on this angle)
* i8tn-friendly (adding @sarina to chime in on this angle as well)

I'd prefer to have this in the DOM and to have the CSS target a UI-based class name to show/hide/transition its visual display.

talbs replied:
If you keep this rule, please address the line-by-line comments below.

sarina replied:
We cannot internationalize out of sass. User facing strings **must** be defined in templates, python, or javascript files.

sarina replied:
It really makes no sense that you're defining content in scss files. This seems very wrong. scss should define styles, templates should define the base views, and python/javascript should define/inject dynamic content.

cptvitamin replied:
I, am course, am with @sarina and @talbs here.  Content, style, behavior; you've got to keep'em separated.  @clrux, we might want to modify this section of the styleguide to include all css generated content, not just BG images:
https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Accessibility#-using-images-dom-or-css

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            @include calc( left, '50% - 100px');
+            content: 'LEARN MORE';
+            position: absolute;
+            width: 200px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable (e.g. ``($baseline*10)``).

talbs replied:
Also, this is another case of using/creating a local Sass variable (e.g. ``$course-image-height` or ``$course-cta-width``).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            content: 'LEARN MORE';
+            position: absolute;
+            width: 200px;
+            height: 50px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable (e.g. ``($baseline*2.5)``).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            position: absolute;
+            width: 200px;
+            height: 50px;
+            z-index: 20;
```
Please use one of our defined UI-depth Sass extends here (for easier UI management down the road) - https://github.com/edx/edx-platform/blob/mjevtic/SOL-531/lms%2Fstatic%2Fsass%2Fbase%2F_mixins.scss#L93

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            width: 200px;
+            height: 50px;
+            z-index: 20;
+            border: 3px solid white;
```
Please use the Sass ``$white`` variable - https://github.com/edx/edx-platform/blob/mjevtic/SOL-531/lms%2Fstatic%2Fsass%2Fbase%2F_variables.scss#L66

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            z-index: 20;
+            border: 3px solid white;
+            border-radius: 3px;
+            font-family: $sans-serif;
```
Are these font-family/weight definitions needed? Shouldn't they be inherited from the parent rule/DOM element?

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            border-radius: 3px;
+            font-family: $sans-serif;
+            font-weight: $font-semibold;
+            font-size: 1.1em;
```
Please use a typography Sass extend if possible when defining font-size/weight - https://github.com/edx/edx-platform/blob/mjevtic/SOL-531/lms%2Fstatic%2Fsass%2Felements%2F_typography.scss.

If you need a specific font-size/setting that the list does not support, please use our ``font-size()`` mixin (e.g. (``@include font-size(18)``)).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            font-family: $sans-serif;
+            font-weight: $font-semibold;
+            font-size: 1.1em;
+            line-height: 44px;
```
If you need a specific line-height, please use our ``line-height()`` mixin by passing in the size of the font you're rendering - the mixin calculates a 1.48 scale value for it (e.g. (``@include line-height(18)``)).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            font-weight: $font-semibold;
+            font-size: 1.1em;
+            line-height: 44px;
+            color: white;
```
Please use the Sass ``$white`` variable - https://github.com/edx/edx-platform/blob/mjevtic/SOL-531/lms%2Fstatic%2Fsass%2Fbase%2F_variables.scss#L66

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            font-size: 1.1em;
+            line-height: 44px;
+            color: white;
+            padding: 0 20px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable (e.g. ``$baseline``).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-          position: relative;
-          width: 100%;
+      .course-info {
+        height: 200px;
```
This is another case of using/creating a local Sass variable (e.g. ``$course-image-height`` or ``$course-cta-width``).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
           img {
-            display: block;
-            min-height: 100%;
             width: 100%;
```
Can this be handled by the ``display: block`` rule you've delete? A block-level element will take up the entire width of its parent's box-model. I'd prefer to have that value vs. the width one.

marjev replied:
The idea here was to maintain the image ratio while constraining it to the width of it's container, because we want to see the full width of the course image the whole time.

Setting `display: block` rule instead didn't allow that. See the image bellow:
![screen shot 2015-04-10 at 10 32 31](https://cloud.githubusercontent.com/assets/6833568/7084465/fb5dc990-df6d-11e4-8112-6a0cce8904cb.png)
Image wasn't presented in the full width.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-            text-overflow: ellipsis;
-          }
+        .course-organization, .course-code, .course-date {
+          color: #3d3e3f;
```
Please use a Sass color variable (https://github.com/edx/edx-platform/blob/mjevtic/SOL-531/lms%2Fstatic%2Fsass%2Fbase%2F_variables.scss) that is already predefined if possible.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-          }
+        .course-organization, .course-code, .course-date {
+          color: #3d3e3f;
+          font-size: 0.9em;
```
Please use a typography Sass extend if possible when defining font-size/weight - https://github.com/edx/edx-platform/blob/mjevtic/SOL-531/lms%2Fstatic%2Fsass%2Felements%2F_typography.scss.

If you need a specific font-size/setting that the list does not support, please use our ``font-size()`` mixin (e.g. (``@include font-size(18)``)).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-          width: 100%;
+        .course-organization {
+          padding: 11px 15px 2px;
+          line-height: 16px;
```
If you need a specific line-height, please use our ``line-height()`` mixin by passing in the size of the font you're rendering - the mixin calculates a 1.48 scale value for it (e.g. (``@include line-height(18)``)).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-          }
+        .course-code {
+          padding: 0 15px;
+          line-height: 24px;
```
If you need a specific line-height, please use our ``line-height()`` mixin by passing in the size of the font you're rendering - the mixin calculates a 1.48 scale value for it (e.g. (``@include line-height(18)``)).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-          padding: 0px 10px 10px 10px;
-          width: 100%;
+        .course-organization {
+          padding: 11px 15px 2px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable.

talbs replied:
Please define all directions when using ``padding`` or ``margin`` shorthand properties.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-            letter-spacing: 1px;
-          }
+        .course-code {
+          padding: 0 15px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+          color: #701f3f;
+          padding: 0 15px;
+          margin: 5px 0;
+          line-height: 24px;
```
If you need a specific line-height, please use our ``line-height()`` mixin by passing in the size of the font you're rendering - the mixin calculates a 1.48 scale value for it (e.g. (``@include line-height(18)``)).

talbs replied:
Please remove if you use a typography extend (noted above).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+
+        .course-date {
+          margin-top: 30px;
+          line-height: 20px;
```
If you need a specific line-height, please use our ``line-height()`` mixin by passing in the size of the font you're rendering - the mixin calculates a 1.48 scale value for it (e.g. (``@include line-height(18)``)).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-          }
+        .course-title {
+          height: 72px;
+          font-size: 1.1em;
```
Please use a typography Sass extend if possible when defining font-size/weight - https://github.com/edx/edx-platform/blob/mjevtic/SOL-531/lms%2Fstatic%2Fsass%2Felements%2F_typography.scss.

If you need a specific font-size/setting that the list does not support, please use our ``font-size()`` mixin (e.g. (``@include font-size(18)``)).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+          height: 72px;
+          font-size: 1.1em;
+          color: #701f3f;
+          padding: 0 15px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+          font-size: 1.1em;
+          color: #701f3f;
+          padding: 0 15px;
+          margin: 5px 0;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-            letter-spacing: 1px;
-          }
+        .course-title {
+          height: 72px;
```
This is another case of using/creating a local Sass variable (e.g. ``$course-title-height``).

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+        }
+
+        .course-date {
+          margin-top: 30px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable.

talbs replied:
Also, we are trying to define our vertical margins in one direction (``margin-bottom``), see - http://csswizardry.com/2012/06/single-direction-margin-declarations/.

Please move this property to another element that can support the ``margin-bottom`` value and keep your intended layout/spacing.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+        .course-date {
+          margin-top: 30px;
+          line-height: 20px;
+          padding: 3px 15px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
         }
       }
 
       &:hover, &:focus {
```
NIT: please add a Sass comment to note the particular interaction state you're styling - https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#sass-comments-for-development

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-        box-shadow: 0 1px 16px 0 rgba($shadow-color, 0.4);
+        .cover-image {
+          &:before {
+            @include fade-transition-delay(0.3s);
```
See previous comment about transitions and update this line.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-        .info {
-          top: -150px;
+          &:after {
+            @include fade-transition-delay(0.3s);
```
See previous comment about transitions and update this line.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+
+      @include media($desktop) {
+        text-align: right;
+        margin-right: 10px;
```
Please, when possible, calculate any spacing values (margin, padding, etc.) using the ``$baseline`` variable.

talbs replied:
RTL Support (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#right-to-left-rtl-support) is needed here.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+      }
+
+      .more-link-arrow:after {
+        content: " ›";
```
Nice abstraction in keeping the decorative/visual `>` element from the DOM! :)

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+      text-align: center;
+      margin-right: 0;
+
+      @include media($desktop) {
```
Are these rules needed for the ``$xl-desktop`` breakpoint as well?

talbs replied:
NIT: property order is out of order (see https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#property-and-declaration-order)

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-          opacity: 0;
-        }
+    .more-courses {
+      text-align: center;
```
NIT: property order is out of order (see https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#property-and-declaration-order)

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+          line-height: 24px;
+        }
+
+        .course-date {
```
NIT: property order is out of order (see https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#property-and-declaration-order)

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-            color: $lighter-base-font-color;
-            letter-spacing: 1px;
-          }
+        .course-title {
```
NIT: property order is out of order (see https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#property-and-declaration-order)

talbs commented on lms/templates/course.html:
```diff
@@ -6,34 +6,17 @@
+      </div>
+    </header>
+    <section class="course-info">
+      <div class="course-organization">${get_course_about_section(course, 'university')}</div>
```
Please use semantic headings to make sure the DOM can be scanned by screen readers and for SEO purposes as well.

talbs replied:
You'll need to adjust some styling if you change the DOM in this way, FYI.

talbs commented on lms/templates/index.html:
```diff
@@ -54,12 +54,17 @@
         <section class="courses">
             <ul class="courses-listing">
-            %for course in courses:
+            %for course in courses[:9]:
```
Is this a cap for only showing 9 or less courses?

talbs commented on lms/templates/index.html:
```diff
@@ -54,12 +54,17 @@
             %endfor
             </ul>
+            <div class="more-courses">
+              <a class="more-link-arrow" title="View all Courses" href="${marketing_link('COURSES')}" >
```
The ``title`` attribute is not needed here since it has the same value as the actual element's text. The text is preferable as it is already i8tn friendly.

cptvitamin replied:
The ```title``` element should really only appear on ```<iframe>``` elements.  I say remove it from here as well.

talbs commented on lms/templates/index.html:
```diff
@@ -54,12 +54,17 @@
               </li>
             %endfor
             </ul>
+            <div class="more-courses">
```
The classes/DOM structure here is a bit weird. I'd prefer to see something like the following instead:

````html
<div class="courses-more">
    <a class="courses-more-cta" href="${marketing_link('COURSES')}" > ... </a>
</div>
````

talbs replied:
You'll need to adjust some styling if you change the DOM in this way, FYI.

talbs commented:
@marjev, thanks for all of the work with this feature. I've made a first pass through the Sass/CSS and HTML/Templates. Please let me know 1) if you have any questions and 2) when you've addressed my comments.

Also, I'm waiting to see this change visually - can you set up a sandbox or is it easier if I pull this locally and start up my LMS using devstack? If the latter is the case, are there any feature flags or settings I need to adjust to see this view?

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -7,33 +7,44 @@
       width: flex-grid(4);
       @include margin-right(flex-gutter());
```
RTL Support (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#right-to-left-rtl-support) is needed here.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
-        }
+    .more-courses {
+      text-align: center;
+      margin-right: 0;
```
RTL Support (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#right-to-left-rtl-support) is needed here.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+      margin-right: 0;
+
+      @include media($desktop) {
+        text-align: right;
```
RTL Support (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#right-to-left-rtl-support) is needed here.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -265,5 +314,6 @@
         margin-right: 0;
```
Not your doing, but RTL Support (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#right-to-left-rtl-support) is needed here.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -7,33 +7,44 @@
       width: flex-grid(4);
       @include margin-right(flex-gutter());
       float: left;
```
Not your doing, but RTL Support (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#right-to-left-rtl-support) is needed here.

talbs commented on lms/static/sass/shared/_course_object.scss:
```diff
@@ -180,78 +191,116 @@
+            content: '';
+            position: absolute;
+            top: 0;
+            left: 0;
```
RTL Support (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#right-to-left-rtl-support) is needed here.

cptvitamin commented on lms/templates/course.html:
```diff
@@ -6,34 +6,17 @@
-    </div>
+    <header class="course-image">
+      <div class="cover-image">
+        <img src="${course_image_url(course)}" alt="${course.display_number_with_default | h} ${get_course_about_section(course, 'title')} Cover Image" />
```
The fact that there is an image here is not important to the SR user, so an empty alt attribute is appropriate ```alt=""```, especially if the image is not going to be described to them (and it mostly likely shouldn't.  These are usually just eye-candy.

cptvitamin commented on lms/templates/course.html:
```diff
@@ -6,34 +6,17 @@
-  %if course.is_newish:
-    <span class="status">${_("New")}</span>
-  %endif
   <a href="${reverse('about_course', args=[course.id.to_deprecated_string()])}">
```
I have an issue with wrapping this anchor element around all of the text associated with the course.  The accessible label for this link will be title of the course, school, number, title, startdate, all read out as one big long string with no breaks or pauses.  In the past, we have included just the course title as the child of the anchor element and expanded the bounds of this element to cover the entire course card.  Alternatively, you can add an aria-hidden=true to the container(s) that wrap the school, number, title and startdate  and then create a separate list of these items that is displayed offscreen using the ```sr``` class.

cptvitamin commented on lms/templates/course.html:
```diff
@@ -6,34 +6,17 @@
 %>
 <%page args="course" />
 <article id="${course.id | h}" class="course">
```
a nice touch here would be to add a ```role="region"``` to this ```article``` element with an ```aria-labelledby``` that references the ID of the ```<h2>``` that has the course title in it.  If you don't want to deal with IDs, you can use ```aria-label="${get_course_about_section(course, 'title')}"```  Which allows a screen reader user to choose whether or not they want to consume the child content of this "region" or to skip over it.

marjev commented:
@talbs I don't have a sandbox instance for this yet. Once I address all the issues commented in here, I'll set it up and let you know.

martynjames commented:
http://pr7548.m.sandbox.edx.org for sandbox

martynjames commented:
jenkins test this please

martynjames commented:
jenkins test this please

marjev commented:
test this please

talbs commented on lms/static/sass/base/_grid-settings.scss:
```diff
@@ -4,10 +4,13 @@
 /* Override the default global box-sizing */
 $border-box-sizing: false;
+/* Grid width variables */
+$large-min-width: 769px;
```
I don't think this variable is needed. It seems to be referenced here, in the ``$large`` breakpoint, and in one particular spot (https://github.com/edx/edx-platform/pull/7548/files#diff-ed9911acad8061e8e3605c66f1d569f4R155). Can we just use the ``$large`` breakpoint instead without this?

marjev replied:
I want the "View all courses" link to be right aligned for min-width when the `$large` breakpoint begins. I don't want that behaviour to be limited by the upper limit of the `$large` breakpoint.
If I do what you've suggested, the link would be aligned in the middle of the parent container for the `huge` breakpoint.

This way, I've only set the `min-width` for the media query in question. To make in more concise, I've used the variable for both media query and the `$large` breakpoint, just in case if we ever changed the lower limit of the `$large` breakpoint.

As always, I'm open to any suggestions on the matter.

talbs commented on lms/static/sass/base/_grid-settings.scss:
```diff
@@ -4,10 +4,13 @@
-$tablet: new-breakpoint(min-width 321px max-width 768px, 8);
-$desktop: new-breakpoint(min-width 769px 12);
-$xl-desktop:  new-breakpoint(min-width 980px 12);
+$tiny: new-breakpoint(max-width 320px 4);
```
Thanks for converting these. Since they are not included in a Sass map which gives context, can you prefix the variables with a ``$bp-*`` to help give context to developers using them? My fear is that someone would use them inappropriately (for font-sizes or other uses).

talbs commented on lms/static/sass/shared/_header.scss:
```diff
@@ -765,7 +765,7 @@ header.global-new {
+    @include media( $large ) {
       .wrapper-header {
         width: 100%;
         min-width: 800px;
```
Not your doing, but could use some clean up (in a separate project/PR by the org).

talbs commented on lms/static/sass/views/_homepage.scss:
```diff
@@ -0,0 +1,168 @@
+  @include outer-container;
+
+  .courses {
+    @include clearfix();
```
Please remove the ``clearfix()`` mixin reference here and replace it with the Neat-based ``row()`` mixin (http://thoughtbot.github.io/neat-docs/latest/#row).

talbs commented on lms/static/sass/views/_homepage.scss:
```diff
@@ -0,0 +1,168 @@
+
+  .courses {
+    @include clearfix();
+    padding: ($baseline*2) 0 ($baseline*0.75) ;
```
NIT: define all four margins in this case.

talbs commented on lms/static/sass/views/_homepage.scss:
```diff
@@ -0,0 +1,168 @@
+    padding: ($baseline*2) 0 ($baseline*0.75) ;
+
+    .courses-listing {
+      margin: 0;
```
Please use the ``%ui-no-list`` extend here (https://github.com/edx/edx-platform/blob/master/lms%2Fstatic%2Fsass%2Fbase%2F_mixins.scss#L133).

talbs commented on lms/static/sass/views/_homepage.scss:
```diff
@@ -0,0 +1,168 @@
+      @include box-sizing(border-box);
+      @include transition(all $tmg-f3 linear 0s);
+      box-shadow: 0 1px 10px 0 $black-t0, inset 0 0 0 1px $white-t3;
+      margin-bottom: ($baseline*1.5);
```
Please move this margin to ``.courses-listing-item`` since it is part of the grid layout and many of the rules around a course in that layout are there.

talbs commented on lms/static/sass/views/_homepage.scss:
```diff
@@ -0,0 +1,168 @@
+      background: $body-bg;
+      width: 100%;
+
+      .course-image {
```
please simplify this selector so that its ``.course-image .cover-image {}`` rather than nesting it.

talbs commented on lms/static/sass/views/_homepage.scss:
```diff
@@ -0,0 +1,168 @@
+          }
+
+          .learn-more {
+            @include box-sizing(border-box);
```
NIT: property order (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#property-and-declaration-order)

marjev replied:
can you please confirm that only `@include box-sizing` and `@include left` are out of order? Other than that, I can't see anything else out of order in here.

talbs commented on lms/static/sass/views/_homepage.scss:
```diff
@@ -0,0 +1,168 @@
+
+          // places the shadow on top of the course image while hovering over the course card
+          &:before {
+            @include left(0);
```
https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#property-and-declaration-order

marjev replied:
can you please specify what is out of order here?

talbs commented on lms/static/sass/views/_homepage.scss:
```diff
@@ -0,0 +1,168 @@
+          margin: ($baseline*0.25) 0 ($baseline*1.75) 0;
+          padding: 0 ($baseline*0.75);
+          height: $course-title-height;
+          color: #701f3f;
```
Please change this value to use the ``$link-color`` platform Sass variable (https://github.com/edx/ux-pattern-library/wiki/Styleguide:-Sass-&-CSS#property-and-declaration-order)

talbs commented on lms/static/sass/views/_homepage.scss:
```diff
@@ -0,0 +1,168 @@
+
+        .course-date {
+          @include line-height(14);
+          padding: ($baseline*0.1) ($baseline*0.75);
```
NIT: please use division (``($baseline/10)``) instead of multiplying by a fraction for the ``padding-top`` and ``padding-bottom`` value here.

talbs commented:
@marjev, thanks for all of the revisions and continued good work. Also, thanks for the sandbox link, @martynjames.

I've made a few inline comments about the current Sass/CSS. Can you please address those.

Also, previewing the UI, there are some visual issues (see screengrabs below):

**No visual padding/margin between course elements and browser window**
![screen shot 2015-04-13 at 9 56 08 am](https://cloud.githubusercontent.com/assets/163763/7117065/54c6be3a-e1c4-11e4-9163-dad79ca376f2.png)

@marjev, can you add the same level of padding/margins that the header has around its elements to the course listings?

- - -

**Header, Welcome Area, and Footer are not responsive**
![screen shot 2015-04-13 at 9 56 14 am](https://cloud.githubusercontent.com/assets/163763/7117059/4f7b95cc-e1c4-11e4-9ad1-d069d588c0ab.png)

![screen shot 2015-04-13 at 10 04 10 am](https://cloud.githubusercontent.com/assets/163763/7117082/71f430a0-e1c4-11e4-8a4b-7fa24f3abaeb.png)

While not part of this PR, these elements are not responsive currently and causing a confusing experience where course cards are flexible while other parts of the same view are not. @marcotuts FYI.

edx-webhook commented:
Thanks for the pull request, @marjev! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request. 

To automatically create an OSPR issue for this pull request, just visit this link: http://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fedx-platform&number=7548

marjev commented:
@talbs please take a look at my latest commit which addresses nits and issues you've commented on in your last review.

Also, I think that it would be a better idea to make a story or series of stories on JIRA regarding the css cleanup (css property reordering, inclusion of existing mixins, variables etc., as well as removal of unused css rules and sass files). I also believe that only course card design should be part of this ticket since I've only got guidelines for course card (responsive) design.

Having a separate JIRA story with guidelines and design for responsiveness of the rest of the homepage (header, footer, title, optional video, etc.) would be a much better approach, IMHO.

FYI, @marcotuts @martynjames 

I would be happy to work on that, once there's an appropriate JIRA story for that.

talbs commented:
Thanks for the update, @marjev. The latest commit looks and you've adressed my immediate feedback/issues.

:+1: 

- - -

I agree that they further responsive clean up work should be done outside of here and am hoping that I or our team can play a more direct role in that in the coming quarter.

@marcotuts, I'm going to schedule a Q4 Open edX Course Discovery pow wow - let's talk about further clean up and visual enhancements.

marcotuts commented:
Sounds great! Thanks @marjev and @talbs

marjev commented:
Since there are two :+1: I'll squash the commits and merge this once the tests pass.
